src/server/snapshot.cc

```diff
@@ -90,79 +90,39 @@ void SliceSnapshot::Start(bool stream_journal, const Cancellation* cll, Snapshot
   snapshot_fb_ = fb2::Fiber("snapshot", [this, stream_journal, cll] {
     IterateBucketsFb(cll, stream_journal);
     db_slice_->UnregisterOnChange(snapshot_version_);
-    if (cll->IsCancelled()) {
-      Cancel();
-    } else if (!stream_journal) {
-      CloseRecordChannel();
+    // We stop the channel if we are performing backups (non-streaming) or full sync failed.
+    // For a successful full-sync we keep the channel in order to switch to streaming mode.
+    if (cll->IsCancelled() || !stream_journal) {
```

Comment:
here is probably the only functional change I did, but if you look at the implementation of Cancel/StopChannel, you will see it fits both usecases.
Now, with the rename its meaning shifted so we can just call the same function.